sirepo/package_data/static/js/activait.js

```diff
@@ -1096,59 +1096,140 @@ SIREPO.app.directive('imagePreviewPanel', function(requestSender) {
+            $scope.ColBName = 'Contour';
+            $scope.predColName = 'Predicted';
+            $scope.imageIdx = 0;
             $scope.dataFileMissing = false;
-            $scope.canUpdateUri = increment => {
-                return $scope.idx + increment >= 0 && $scope.idx + increment < numPages;
+            $scope.hasThirdColumn = true;
+
+            const pageIndex = () => {
+                return $scope.imageIdx / $scope.imagesPerPage;
+            };
+
+            $scope.page = () => Math.floor($scope.imageIdx / $scope.imagesPerPage) + 1;
+
+            $scope.canUpdateUri = (increment) => {
+                return $scope.imageIdx + increment >= 0 && pageIndex() + increment < $scope.numPages;
             };
 
             $scope.first = () => {
-                setImageFromUriIndex($scope.idx = 0);
+                setIndex($scope.imageIdx = 0);
             };
 
             $scope.isLoading = () => loading;
 
             $scope.last = () => {
-                setImageFromUriIndex($scope.idx = $scope.uris.length - 1);
+                setIndex($scope.imageIdx = $scope.numPages * $scope.imagesPerPage - $scope.imagesPerPage);
             };
 
             $scope.next = () => {
-                setImageFromUriIndex($scope.idx += 1);
+                setIndex($scope.imageIdx += $scope.imagesPerPage);
             };
 
             $scope.prev = () => {
-                setImageFromUriIndex($scope.idx -= 1);
+                setIndex($scope.imageIdx -= $scope.imagesPerPage);
             };
 
-            function setImageFromUriIndex(index) {
-                if ($('.' + $scope.imageClass).length && $scope.uris) {
-                    $('.' + $scope.imageClass)[0].src = $scope.uris[index];
+            $scope.colClass = () => `col-md-${$scope.hasThirdColumn ? '4' : '6'}`;
+
+            function imageInRange(firstImageIndex, rowIndex) {
+                return firstImageIndex + rowIndex + 1 <= $scope.colA.length;
+            }
+
+            function setColumnImage(firstImageIndex, rowIndex, column) {
+                let isTextCol = column == 'A' ? $scope.xIsParams : $scope.imageToLabels;
+                let textCol = column == 'A' ? $scope.parameters : $scope.labels;
+                if (isTextCol) {
+                    if (imageInRange(firstImageIndex, rowIndex)) {
+                        let value = column == 'A' ? `${$scope.colA[firstImageIndex + rowIndex].replace(/[\[\]]/g, '')}` : $scope.colB[firstImageIndex + rowIndex];
+                        textCol.splice(rowIndex, 0, value);
+                        return;
+                    }
+                    textCol.splice(rowIndex, 0, '');
+                    return;
+                }
+                if (imageInRange(firstImageIndex, rowIndex)) {
+                    let value = column == 'A' ? $scope.colA[firstImageIndex + rowIndex] : $scope.colB[firstImageIndex + rowIndex];
+                    $(`.col${column}${$scope.method}${rowIndex + 1}`)[0].src = value;
```

Comment:
Use a constant for the jquery selector. Also, restrict the selector to this element:
```javascript
const colSelector = $($element).find(`.col${column}${$scope.method}${rowIndex + 1}`)[0];
```